src/components/button/createButton.ts

```diff
@@ -2,33 +2,64 @@ import { ComposeFactory } from 'dojo-compose/compose';
 import { VNodeProperties } from 'dojo-interfaces/vdom';
 import createWidgetBase from '../../createWidgetBase';
 import { Widget, WidgetOptions, WidgetProperties, WidgetState } from './../../interfaces';
-import createFormFieldMixin, { FormFieldMixin, FormFieldMixinState, FormFieldMixinOptions } from '../../mixins/createFormFieldMixin';
 
-export interface ButtonState extends WidgetState, FormFieldMixinState<string> {
-	label?: string;
+export interface ButtonState extends WidgetState {
+	/*
+	 * Text content of button
+	 */
+	content?: string;
+
+	/**
+	 * The button's name
+	 */
+	name?: string;
+
+	/**
+	 * Accessibility attributes
+	 */
+	disabled?: boolean;
+	pressed?: boolean;
 }
 
 export interface ButtonProperties extends WidgetProperties {
-	label?: string;
+	content?: string;
+	name?: string;
+	type?: string;
+
+	/**
+	 * Accessibility attributes
+	 */
+	descriptionID?: string;
+	disabled?: boolean;
+	role?: string;
+	pressed?: boolean;
+	hasPopup?: boolean;
 }
 
-export interface ButtonOptions extends WidgetOptions<ButtonState, ButtonProperties>, FormFieldMixinOptions<any, ButtonState> { }
+export interface ButtonOptions extends WidgetOptions<ButtonState, ButtonProperties> { }
 
-export type Button = Widget<ButtonState, ButtonProperties> & FormFieldMixin<string, ButtonState>;
+export type Button = Widget<ButtonState, ButtonProperties>;
 
 export interface ButtonFactory extends ComposeFactory<Button, ButtonOptions> { }
 
 const createButton: ButtonFactory = createWidgetBase
-	.mixin(createFormFieldMixin)
 	.mixin({
 		mixin: {
 			nodeAttributes: [
 				function(this: Button): VNodeProperties {
-					return { innerHTML: this.state.label };
+					return {
+						innerHTML: this.properties['content'],
+						type: this.properties['type'],
+						name: this.properties['name'],
+						role: this.properties['role'],
+						disabled: this.state.disabled,
+						'aria-pressed': this.state.pressed,
```

Comment:
There are some attributes like `name`, `role`, `type`, and `aria-describedby` that I don't think should be updated after creation. In my opinion it would be unnecessarily confusing to assistive tech, and it's hard to think of a valid use case. If you think it goes against the pattern for creating widgets, though, I can change it to access them through `state`.